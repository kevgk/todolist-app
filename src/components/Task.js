import React from 'react';
import TaskCheckbox from './TaskCheckbox';

export default function Task({ name, clickHandler, removeHandler, isChecked }) {
	return (
		<li className='task'>
			<TaskCheckbox isChecked={isChecked} clickHandler={clickHandler} />
			<span className='taskname'>{name}</span>
			{isChecked && (
				<button className='removeButton' onClick={() => removeHandler()}>
					x
				</button>
			)}
		</li>
	);
}
