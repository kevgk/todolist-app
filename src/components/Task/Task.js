import React from 'react';
import styles from './Task.module.css';
import TaskCheckbox from '../TaskCheckbox/TaskCheckbox';

export default function Task({ name, clickHandler, removeHandler, isChecked }) {
  return (
    <li className={styles.task}>
      <TaskCheckbox isChecked={isChecked} clickHandler={clickHandler}/>
      <span className={styles.taskname}>{name}</span>
      {isChecked && <button className={styles.removeButton} onClick={() => removeHandler()}>x</button>}
    </li>
  );
}