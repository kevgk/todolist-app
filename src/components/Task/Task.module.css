.task {
  display: flex;
  cursor: pointer;
  background-color: #eee;
  padding: 7px 10px;
  border-radius: 10px;
  font-size: 14px;
  margin-bottom: 10px;
  transition: background-color 0.3s;
}

.task:hover {
  background-color: #e9e9e9;
}

.taskname {
  flex-grow: 1;
  padding-left: 10px;
  position: relative;
  top: 2px;
}

@keyframes pulse {
  0% { transform: scale(1) }
  50% { transform: scale(1.2) }
  100% { transform: scale(1) }
}

.checkbox {
  width: 20px;
  height: 20px;
  border: none;
  border-radius: 5px;
  background-color: #7047f5;
  cursor: pointer;
  display: flex;
  justify-content: center;
}

.task:hover .checkbox {
  animation-name: pulse;
  animation-duration: 0.6s;
  animation-iteration-count: infinite;
}

.checkbox > span {
  transition: opacity 0.3s;
  color: #fff;
}

.removeButton {
  border: none;
  background-color: transparent;
}