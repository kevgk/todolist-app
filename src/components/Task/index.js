import React from 'react';
import styles from './task.module.css';

export default function Task({ name, checkHandler, removeHandler, isChecked }) {
  return (
    <li className={styles.task}>
      <button className={styles.checkbox} onClick={() => checkHandler()}><span role="img" aria-label="check task" style={{ opacity: isChecked ? 1 : 0.1}}>âœ”</span></button>
      <span className={styles.taskname}>{name}</span>
      {isChecked && <button className={styles.removeButton} onClick={() => removeHandler()}>x</button>}
    </li>
  );
}