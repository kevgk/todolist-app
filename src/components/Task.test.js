import React from 'react';
import { mount } from 'enzyme';

import Task from './Task';

it('renders correctly (unchecked)', () => {
	const wrapper = mount(
		<Task
			name='Test Task'
			clickHandler={() => {}}
			removeHandler={() => {}}
			isChecked={false}
		/>
	);

	expect(wrapper).toMatchSnapshot();
});

it('renders correctly (checked)', () => {
	const wrapper = mount(
		<Task
			name='Test Task'
			clickHandler={() => {}}
			removeHandler={() => {}}
			isChecked={true}
		/>
	);

	expect(wrapper).toMatchSnapshot();
});
