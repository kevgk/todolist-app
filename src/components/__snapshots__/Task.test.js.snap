// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly (checked) 1`] = `
<Memo(Task)
  clickHandler={[Function]}
  isChecked={true}
  name="Test Task"
  removeHandler={[Function]}
>
  <li
    className="task"
  >
    <button
      className="checkbox"
      onClick={[Function]}
    >
      <span
        aria-label="check task"
        role="img"
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        ✔
      </span>
    </button>
    <span
      className="taskname"
    >
      Test Task
    </span>
    <button
      className="removeButton"
      onClick={[Function]}
    >
      x
    </button>
  </li>
</Memo(Task)>
`;

exports[`renders correctly (unchecked) 1`] = `
<Memo(Task)
  clickHandler={[Function]}
  isChecked={false}
  name="Test Task"
  removeHandler={[Function]}
>
  <li
    className="task"
  >
    <button
      className="checkbox"
      onClick={[Function]}
    >
      <span
        aria-label="check task"
        role="img"
        style={
          Object {
            "opacity": 0.1,
          }
        }
      >
        ✔
      </span>
    </button>
    <span
      className="taskname"
    >
      Test Task
    </span>
  </li>
</Memo(Task)>
`;
